src/ChanperfTracker.ts

```diff
@@ -0,0 +1,145 @@
+import { EventEmitter } from 'events';
+
+export class ChanperfTracker {
+    constructor() {
+        this.emitter = new EventEmitter();
+        this.filterOutLogs = true;
+        this.chanperfHistory = [];
+    }
+
+    private emitter: EventEmitter;
+    private filterOutLogs: boolean;
+    private chanperfHistory: Array<ChanperfEventData>;
+
+    public on(eventname: 'chanperf-event', handler: (output: ChanperfEventData) => void);
+    public on(eventName: string, handler: (payload: any) => void) {
+        this.emitter.on(eventName, handler);
+        return () => {
+            if (this.emitter !== undefined) {
+                this.emitter.removeListener(eventName, handler);
+            }
+        };
+    }
+
+    private emit(eventName: 'chanperf-event', data: ChanperfEventData) {
+        this.emitter.emit(eventName, data);
+    }
+
+    public get getChanperfHistory(): Array<ChanperfEventData> {
+        return this.chanperfHistory;
+    }
+
+    /**
+     * Used to set wether the chanperf should be filtered from the console output
+     * @param outputLevel the consoleOutput from the launch config
+     */
+    public setConsoleOutput(outputLevel: string) {
+        this.filterOutLogs = (outputLevel !== 'full');
+    }
+
+    /**
+     * Clears the current chanperf history
+     */
+    public clearChanperfHistory() {
```

Comment:
You don't need the word "Chanperf" in this. It's the `ChanperfTracker` class, so `clearHistory` will do just fine.